v3/pkg/application/messageprocessor_call.go

```diff
@@ -4,19 +4,16 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
-	"strconv"
 )
 
 func (m *MessageProcessor) callErrorCallback(window *WebviewWindow, message string, callID *string, err error) {
 	errorMsg := fmt.Sprintf(message, err)
 	m.Error(errorMsg)
-	msg := "_wails.callErrorCallback('" + *callID + "', " + strconv.Quote(errorMsg) + ");"
-	window.ExecJS(msg)
+	window.CallError(callID, errorMsg)
 }
 
 func (m *MessageProcessor) callCallback(window *WebviewWindow, callID *string, result string, isJSON bool) {
-	msg := fmt.Sprintf("_wails.callCallback('%s', %s, %v);", *callID, strconv.Quote(result), isJSON)
```

Comment:
I think if we can resolve the `isJSON` question then we could get rid of this method and call it on the window directly.

Reply:
That's a good point.  Everywhere the call existed it was passing a `true` already so I think might should create a `type` that enforces the object passed in to have a `ToJSON` function implemented in order to be acceptable. 
What do you think of that?

Reply:
I think we should be able to get away with parsing it as json and if it fails just return the raw value. I'll try looking into this tomorrow 👍

Reply:
Just as an aside. The Dialogs work just fine with this code.  They always return a string, so it seems it's OK. Although I think we should update such that you can't possibly put in a bad value.